lib/packages/Pattern/SingletonDecorator.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace WooCommerce\PayPalCommerce\Vendor\Pattern;
```

Comment:
Is this class written by you? Not sure if this dir is a good place for such things, it was just a quick fix for isolating some deps that can cause conflicts if there are other plugins with different versions of the same dep.

Maybe it should be some separate subdir (like `lib/isolated` and `lib/packages`) with some less weird namespace for non-generated packages then.

Reply:
Yep I did it. It doesn't really fit any module it's too generic.

```
        "psr-4": {
            "WooCommerce\\PayPalCommerce\\": "src",
            "WooCommerce\\PayPalCommerce\\Vendor\\": "lib/packages/"
        },
```

Maybe have also a:
"WooCommerce\\PayPalCommerce\\Common\\": "lib/common/"

For these cases or `lib/isolated`